tests/test_core.py

```diff
@@ -4122,17 +4122,17 @@ def test_getdents64(self):
     src = open(path_from_root('tests', 'fs', 'test_getdents64.cpp'), 'r').read()
     self.do_run(src, '..')
 
+  def test_getdents64_special_cases(self):
+    Building.COMPILER_TEST_OPTS += ['-std=c++11']
+    src = path_from_root('tests', 'fs', 'test_getdents64_special_cases.cpp')
+    out = path_from_root('tests', 'fs', 'test_getdents64_special_cases.out')
+    self.do_run_from_file(src, out, assert_identical=True)
```

Comment:
FYI: assert_identical is needed here because otherwise the line "doTest("test_dir3", std::string(254, '1') + u8"今日は");" would succeeded even if I deliberately broke stringToUTF8Array.

Reply:
Reading the changes to the test harness, introducing `assert_identical=True` looks ok to me, though @kripken, would be good for you to also take a glance for that part?